internal/cmd/provider/goreleaser.go

```diff
@@ -0,0 +1,219 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// GoReleaserVersionData contains the data we get from GoReleaser's distribution
+// directory.
+type GoReleaserVersionData struct {
+	Artifacts GoReleaserArtifacts
+	Metadata  GoReleaserMetadata
+	Changelog *string
+}
+
+// BuildGoReleaserVersionData builds a GoReleaserMetadata struct by looking at the
+// distribution directory created by GoReleaser.
+func BuildGoReleaserVersionData(dir string) (*GoReleaserVersionData, error) {
+	var out GoReleaserVersionData
+
+	// Read the artifacts file.
+	artifactsPath := dir + "/artifacts.json"
+
+	// #nosec G304
+	artifactsData, err := os.ReadFile(artifactsPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read %s", artifactsPath)
+	}
+
+	if err := json.Unmarshal(artifactsData, &out.Artifacts); err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal %s", artifactsPath)
+	}
+
+	// Read the metadata file.
+	metadataPath := dir + "/metadata.json"
```

Comment:
It's safer to use filepath.Join as dir might be a custom value provided by the user